src/bodse/avl/raw.py

```diff
@@ -0,0 +1,322 @@
+
+    response_time: dt.datetime
+    producer_ref: str
+    valid_until: dt.datetime
+    shortest_cycle: str
+
+    _producer_ref_tag = "siri:ServiceDelivery/siri:ProducerRef"
+    _vehicle_monitoring_tags = {
+        "response_time": "siri:ResponseTimestamp",
+        "valid_until": "siri:ValidUntil",
+        "shortest_cycle": "siri:ShortestPossibleCycle",
+    }
+
+    @classmethod
+    def parse_xml(cls, root: ET.Element) -> AVLMetadata:
+        """Parse and extract metadata from Siri XML data.
+
+        Parameters
+        ----------
+        root: ET.Element
+            Root XML element from Siri data.
+        """
+        parent = "AVLMetadata"
+
+        producer_ref = _findtext(root, cls._producer_ref_tag, parent, default="unknown")
+        assert isinstance(producer_ref, str)
+
+        vehicle_monitoring = _find(root, VEHICLE_MONITORING_TAG, parent, if_missing="error")
+        assert vehicle_monitoring is not None
+
+        data = {}
+        for name, tag in cls._vehicle_monitoring_tags.items():
+            data[name] = _findtext(vehicle_monitoring, tag, parent, if_missing="error")
+
+        return AVLMetadata(producer_ref=producer_ref, **data)
+
+    @classmethod
+    def get_fields(cls) -> list[fields.ModelField]:
+        """List of data fields."""
+        # Added by dataclass decorator pylint: disable=no-member
+        return list(cls.__dataclass_fields__.values())
+
+    @property
+    def data_dict(self) -> dict[str, Any]:
+        """Convert class to dictionary."""
+        return {i.name: getattr(self, i.name) for i in self.get_fields()}
+
+
+@dataclasses.dataclass
+class VehicleActivity:
+    """AVL vehicle activity data."""
+
+    xml_item_id: str
+    recorded: dt.datetime
+    valid_until: dt.datetime
+    longitude: float
+    latitude: float
+
+    bearing: Optional[float] = None
+    line_ref: Optional[str] = None
+    direction: Optional[str] = None
+    line_name: Optional[str] = None
+    operator: Optional[str] = None
+    vehicle_ref: Optional[str] = None
+    origin_ref: Optional[str] = None
+    destination_ref: Optional[str] = None
+    block_ref: Optional[str] = None
+    origin_name: Optional[str] = None
+    destination_name: Optional[str] = None
+    occupancy: Optional[str] = None
+    origin_scheduled_time: Optional[dt.datetime] = None
+    destination_scheduled_time: Optional[dt.datetime] = None
+    # Optional framed vehicle journey info
+    data_frame_ref: Optional[str] = None
+    vehicle_journey_ref: Optional[str] = None
+
+    # Optional tags found in extensions
+    vehicle_id: Optional[str] = None
+    seated_occupancy: Optional[int] = None
+    seated_capacity: Optional[int] = None
+    wheelchair_occupancy: Optional[int] = None
+    wheelchair_capacity: Optional[int] = None
+    occupancy_threshold: Optional[str] = None
+    ticket_machine_service_code: Optional[str] = None
+    journey_code: Optional[str] = None
+
+    # XML tag lookups
+    _expected_tags = {
```

Comment:
This file isn't actually used anymore, it was part of some tests looking at using the TransXChange format instead of GTFS. It was partially working for parsing the TXC format but we decided against using it while it was still WIP, might be worth me making that clearer in the module docstring that it's WIP and no currently used.